compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -2734,40 +2735,52 @@ impl<'a> Parser<'a> {
             if self.token.kind == TokenKind::Eof {
                 break;
             }
-            if let Some(span) = self.diff_marker(&TokenKind::OrOr, &TokenKind::BinOp(token::Or)) {
+            if let Some(span) = self.conflict_marker(&TokenKind::OrOr, &TokenKind::BinOp(token::Or))
+            {
                 middlediff3 = Some(span);
             }
-            if let Some(span) = self.diff_marker(&TokenKind::EqEq, &TokenKind::Eq) {
+            if let Some(span) = self.conflict_marker(&TokenKind::EqEq, &TokenKind::Eq) {
                 middle = Some(span);
             }
-            if let Some(span) = self.diff_marker(&TokenKind::BinOp(token::Shr), &TokenKind::Gt) {
+            if let Some(span) = self.conflict_marker(&TokenKind::BinOp(token::Shr), &TokenKind::Gt)
+            {
                 spans.push(span);
                 end = Some(span);
                 break;
             }
             self.bump();
         }
-        let mut err = self.struct_span_err(spans, "encountered diff marker");
-        err.span_label(start, "after this is the code before the merge");
+        let mut err = self.struct_span_err(spans, "encountered git conflict marker");
```

Comment:
`git` is not the only application that uses these kinds of conflict markers.

Reply:
Hmm that's true. What about the "for more information, visit the `git` documentation" note at the end, should that perhaps be reworded to something more along the lines of "if you're using `git` check out this documentation"?

Reply:
Sounds good to me (although I feel that `git`'s docs are useful even if you're not using `git` specifically).